packages/pipeline/src/parsers/copper/index.ts

```diff
@@ -0,0 +1,274 @@
+        entity.id = activity.id;
+
+        entity.parentId = activity.parent.id;
+        entity.parentType = activity.parent.type;
+
+        entity.typeId = activity.type.id;
+        entity.typeCategory = activity.type.category.toString();
+        entity.typeName = activity.type.name;
+
+        entity.userId = activity.user_id;
+        entity.dateCreated = activity.date_created * ONE_SECOND;
+        entity.dateModified = activity.date_modified * ONE_SECOND;
+
+        // nullable fields
+        try {
+            entity.oldValueId = activity.old_value.id;
+            entity.oldValueName = activity.old_value.name;
+            entity.newValueId = activity.new_value.id;
+            entity.newValueName = activity.new_value.name;
+        } catch (e) {
+            return entity;
+        }
+        return entity;
+    });
+}
+
+/**
+ * Parse response from Copper API /search/opportunities/
+ *
+ * @param opportunities - The array of opportunities returned from the API
+ * @returns Returns an array of Copper Opportunity entities
+ */
+export function parseOpportunities(opportunities: CopperOpportunityResponse[]): CopperOpportunity[] {
+    return opportunities.map(opp => {
+        const entity = new CopperOpportunity();
+        entity.id = opp.id;
+        entity.name = opp.name;
+        entity.assigneeId = opp.assignee_id;
+        entity.closeDate = opp.close_date;
+        entity.companyId = opp.company_id;
+        entity.companyName = opp.company_name;
+        entity.customerSourceId = opp.customer_source_id;
+        entity.lossReasonId = opp.loss_reason_id;
+        entity.pipelineId = opp.pipeline_id;
+        entity.pipelineStageId = opp.pipeline_stage_id;
+        entity.primaryContactId = opp.primary_contact_id;
+        entity.priority = opp.priority;
+        entity.status = opp.status;
+        entity.tags = opp.tags;
+        entity.interactionCount = opp.interaction_count;
+        entity.monetaryValue = opp.monetary_value;
+        entity.winProbability = opp.win_probability;
+        entity.dateCreated = opp.date_created * ONE_SECOND;
+        entity.dateModified = opp.date_modified * ONE_SECOND;
+        entity.customFields = opp.custom_fields.map(f => f.custom_field_definition_id);
+        return entity;
+    });
+}
+
+/**
+ * Parse response from Copper API /activity_types/
+ *
+ * @param activityTypeResponse - Activity Types response from the API, keyed by "user" or "system"
+ * @returns Returns an array of Copper Activity Type entities
+ */
+export function parseActivityTypes(
+    activityTypeResponse: Map<CopperActivityTypeCategory, CopperActivityTypeResponse[]>,
+): CopperActivityType[] {
+    const values: CopperActivityTypeResponse[] = R.reduce(
+        (acc, val) => acc.concat(val),
+        [] as CopperActivityTypeResponse[],
+        Object.values(activityTypeResponse),
+    );
+    return values.map(activityType => {
+        const parsed: CopperActivityType = {
```

Comment:
nit: I think we already have enough context here to understand what's going on without declaring the temporary variable called `parsed`. (The function is called `parseActivityTypes`).

Can we do this instead?

```ts
values.map(activityType => ({
    id: activityType.id,
    name: activityType.name,
    category: activityType.category.toString(),
    isDisabled: activityType.is_disabled,
    countAsInteraction: activityType.count_as_interaction,
}));
```